homeassistant/components/knx/cover.py

```diff
@@ -113,6 +113,13 @@ def is_closing(self) -> bool:
         """Return if the cover is closing or not."""
         return self._device.is_closing()
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return device specific state attributes."""
+        if self._device.supports_locked:
+            return {ATTR_MOVEMENT_LOCKED: self._device.is_locked()}
```

Comment:
This is a state hidden in an attribute and should be its own entity. In this case a binary sensor.